experimental/nomad_resource_plugin/licensedevice/cmd/docker_events/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/enfabrica/enkit/experimental/nomad_resource_plugin/licensedevice/docker"
+)
+
+func exitIf(err error) {
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	c, err := docker.NewClient(ctx)
+	exitIf(err)
+
+	eventsChan := c.Chan(ctx)
+
+	fmt.Println("Listening for events...")
+
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop
```

Comment:
unless im missing something...

the jump back to `loop` will cause this case to get called again no? and at this point it would either cause an infinite loop (assuming the Done() channel is closed), or never reach this case again because the Done() channel is drained.